Add NAT to logs. Show in new view-logs page.

lucyhe opened the pull request:
Checking the checkbox opens this dialog:
Clicking More Info takes you to the "does uproxy log data about me" question in the FAQ.
![image](https://cloud.githubusercontent.com/assets/8723127/6875848/bc90201c-d498-11e4-9f42-d83787ce97ae.png)


I updated the hover text:
![image](https://cloud.githubusercontent.com/assets/8723127/6875326/f0244caa-d493-11e4-9337-a1db4fff4b88.png)

Clicking "Preview" opens this:
Clicking "No" just sends you back to the feedback page, more info goes to FAQ and yes will open the new view logs page (below):
![image](https://cloud.githubusercontent.com/assets/8723127/6875840/af1ad1fc-d498-11e4-83ad-9233b0751ccd.png)


View logs page, while loading:
![image](https://cloud.githubusercontent.com/assets/8723127/6875136/33832e40-d493-11e4-9bf4-a2525031ed96.png)

View logs page, loaded, with NAT data at the top:
![image](https://cloud.githubusercontent.com/assets/8723127/6878466/06c3de5a-d4b5-11e4-966b-eeedcedc2e98.png)







<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/uproxy/uproxy/1173)
<!-- Reviewable:end -->

lucyhe commented:
Updated feedback tooltip, checkbox and PII text:
![image](https://cloud.githubusercontent.com/assets/8723127/6879486/3a35558a-d4cd-11e4-90e0-cd83c33e559a.png)

lucyhe commented:
To explain the dialogs a little more:
User logs are a reflection of uProxy activity -- this might include PII, which we warn users about below the checkbox. In contrast, running network diagnostics involves behaviour outside of the regular scope of using uProxy. Because of that, it seems right to also get user consent to run something that: 1) they would not expect a proxy service to do by default and 2) that might reveal their IP (and potentially other info? @iislucas @gwpenn?) With this rationale, I added the first dialog (with the checkbox) so that users understand clicking submit will start running the diagnostics. Similarly, if users want to preview their logs/network info, as soon as they confirm to preview, the diagnostics start to run. I worry that if a user clicks the icon and are presented with a page that says "running diagnostics...", they may be surprised. Lucas has also suggested putting a button on the "view logs" page to run diagnostics, which I think is a good alternative to the dialog.
In other words, the "preview" dialog is asking users to confirm whether it's okay for the diagnostics to start running. Even if they opt not to submit the network info and logs, we would already have run the NAT test so that they could see the results, and we want users to be aware of the fact that they're running something extra just by previewing.

Sample of extra logs that would exist because of NAT classifying:
![image](https://cloud.githubusercontent.com/assets/8723127/6879510/5923e366-d4ce-11e4-9529-15b536124cc2.png)

lucyhe commented:
Updated the checkbox/hover text to action words:
![image](https://cloud.githubusercontent.com/assets/8723127/6898232/e6a04d48-d6c6-11e4-8271-a499ad975d7d.png)
The help icon beside logs goes to "Does uProxy log data about me". This should be replaced with a more relevant question.

lucyhe commented:
Quick note: I don't believe that diagnose.ts and messages.ts need to be reviewed

salomegeo commented:
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/1173)

Reviewed files:
- src/generic_core/core.ts @ r3
- src/generic_core/diagnose.ts @ r3
- src/generic_core/freedom-module.json @ r2
- src/generic_core/messages.ts @ r1
- src/generic_ui/polymer/feedback.html @ r3
- src/generic_ui/polymer/feedback.ts @ r3
- src/generic_ui/polymer/logs.html @ r4
- src/generic_ui/polymer/logs.ts @ r1
- src/generic_ui/polymer/root.html @ r1
- src/generic_ui/view-logs.html @ r1
- src/uproxy.ts @ r2





---

<sup>**[src/generic_core/core.ts, line 444 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfU-SdD1dSPSrektou)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_core/core.ts#L444)):</sup>
Do we need to check in this comment?

---

<sup>**[src/generic_core/core.ts, line 480 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfVKL5bq_aaTdIO70x)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_core/core.ts#L480)):</sup>
This too

---

<sup>**[src/generic_core/core.ts, line 511 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfVbbXmBCJ5hcQu2fo)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_core/core.ts#L511)):</sup>
How often do you think users submit feedback?

User might do this, try something, then change the internet and try it again, in that case they are submitting wrong feedback.

I don't think too strongly about this, but I think querying every time to get the nat type should be affordable.

---

<sup>**[src/generic_ui/polymer/logs.ts, line 14 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfT8q8JPsLoWgAbvCI)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_ui/polymer/logs.ts#L14)):</sup>
Why do you need this statement?

---

<sup>**[src/generic_ui/view-logs.html, line 6 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfSYzvWcCuCz7OwFk7)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_ui/view-logs.html#L6)):</sup>
is there a way to avoid including dependencies from here?

---


<!-- Sent from Reviewable.io -->

lucyhe commented:
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/1173)







---

<sup>**[src/generic_core/core.ts, line 444 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfU-SdD1dSPSrektou)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_core/core.ts#L444)):</sup>
Removed, will add in new PR

---

<sup>**[src/generic_core/core.ts, line 480 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfVKL5bq_aaTdIO70x)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_core/core.ts#L480)):</sup>
Removed, will add in new PR

---

<sup>**[src/generic_core/core.ts, line 511 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfVbbXmBCJ5hcQu2fo)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_core/core.ts#L511)):</sup>
It gets pretty slow so I think caching is good if 1) they open "analyze network" more than once without changing any of their settings or 2) if they submit right after viewing their logs. I think your point about them changing the internet makes a lot of sense. If we end up informing users about NAT types, and they see that they're on a symmetric NAT, user's should be able to re-run diagnostics at will. So if they switch internet, it seems quite bad to show them the same result...

---

<sup>**[src/generic_ui/polymer/logs.ts, line 14 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfT8q8JPsLoWgAbvCI)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_ui/polymer/logs.ts#L14)):</sup>
Ah I don't. Removed!

---

<sup>**[src/generic_ui/view-logs.html, line 6 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfSYzvWcCuCz7OwFk7)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_ui/view-logs.html#L6)):</sup>
I can replace with 

  <script src='scripts/user.js'></script>
  <script src='scripts/ui.js'></script>
  <script src='scripts/dependencies.js'></script>

will that help a little?

---


<!-- Sent from Reviewable.io -->

lucyhe commented:
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/1173)







---

<sup>**[src/generic_ui/view-logs.html, line 6 \[r4\]](https://reviewable.io:443/reviews/uproxy/uproxy/1173#-JlfSYzvWcCuCz7OwFk7)** ([raw file](https://github.com/uproxy/uproxy/blob/896e87bf90c937bce67ab9961f8a32b4637fdd12/src/generic_ui/view-logs.html#L6)):</sup>
Sorry, did not format correctly, I meant replacing with this:
```HTML
<script src='scripts/user.js'></script>
<script src='scripts/ui.js'></script>
<script src='scripts/dependencies.js'></script>
```

---


<!-- Sent from Reviewable.io -->

salomegeo commented:
:+1:
